### ISSUE TITLE
Exception when attempting to get the next filename left tablet in bad state

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#1298]
A user encountered an issue where Hadoop threw a RemoteException when getNextMapFilename was called at [MinorCompactionTask.java line 62](https://github.com/apache/accumulo/blob/rel/1.9.3/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/MinorCompactionTask.java#L62).  Below is a snippet of the stack trace.

```
VolumeManagerImpl.listStatus(...)    
Tablet.checkTabletDir(Tablet.java:286)
Tablet.getNextMapFilename(Tablet.java:277)
MinorCompactionTask.run(MinorCompactionTask.java:62)
```

This code does not retry on IOException.  When it threw an exception it left the tablet in a bad state where it would never complete the minor compaction.  The `Tablet.checkTabletDir()` method accessed DFS causing this issue.

**Comment by milleruntime** [comment]
Do you think this is related to #1646 ? That error was different but it sounds like a similar situation.

**Comment by keith-turner** [comment]
@milleruntime I will look into this.

**Comment by milleruntime** [comment]
@keith-turner I am looking at this one for 1.10.  I think we can fix it with some simple refactoring.

**Comment by milleruntime** [comment]
I think I have a simple fix with #1675 that shouldn't be a problem merging forward

**Comment by keith-turner** [comment]
@milleruntime I looked #1675, it looks great and really like the simplicity of the fix.  Looking over it inspired me to open #1683.
